### PR TITLE
Cart update and checkout validation for cart

### PR DESCRIPTION
**Pull request by BabovicT**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 

It was possible to update the cart from cart summary view without form validation being triggered. For some reason cart validation is not working there but new Order one is. 
Also there was no cart validation on first checkout step and you were able to order more products than there is.

### CODE REVIEW

File: src/CoreShop/Bundle/CoreBundle/Checkout/Step/CustomerCheckoutStep.php [hunk 1]
```diff
@@ -17,6 +17,7 @@
 use CoreShop\Bundle\CoreBundle\Customer\CustomerManagerInterface;
 use CoreShop\Bundle\CoreBundle\Form\Type\GuestRegistrationType;
 use CoreShop\Component\Core\Model\CustomerInterface;
+use CoreShop\Component\Inventory\Checker\AvailabilityCheckerInterface;
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/CoreBundle/Checkout/Step/CustomerCheckoutStep.php:20]: not needed

**Review by dpfaffenbauer — CHANGES_REQUESTED** [review]

File: src/CoreShop/Bundle/FrontendBundle/Controller/CheckoutController.php [hunk 2]
```diff
@@ -45,6 +47,12 @@ public function processAction(Request $request): Response
             return $this->redirectToRoute('coreshop_cart_summary');
         }
 
+        foreach ($this->getCart()->getItems() as $item) {
+            if (!$this->availabilityChecker->isStockSufficient($item->getProduct(), 0)) {
```
[the remark this answers is not in the document]
Comment (BabovicT) [reply src/CoreShop/Bundle/FrontendBundle/Controller/CheckoutController.php:51]: I will remove it. Not necessary anymore

File: src/CoreShop/Bundle/FrontendBundle/Controller/CartController.php [hunk 3]
```diff
@@ -64,7 +65,11 @@ public function createQuoteAction(Request $request, StateMachineManagerInterface
     public function summaryAction(Request $request): Response
     {
         $cart = $this->getCart();
+        $cloneCart = clone($cart);
```
[the remark this answers is not in the document]
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/FrontendBundle/Controller/CartController.php:68]: Problem is that validation fails cart shows wrong information?

[the remark this answers is not in the document]
Comment (BabovicT) [reply src/CoreShop/Bundle/FrontendBundle/Controller/CartController.php:68]: Yes. If we have 1 item in cart and that 1 is last item available and we try to add one more then:
If there is no form copy cart will show 2.
If there is form copy cart will show 1.
In both cases error is shown and cart resets to 1 after submit and going to next step.

[the remark this answers is not in the document]
Comment (BabovicT) [reply src/CoreShop/Bundle/FrontendBundle/Controller/CartController.php:68]: Cart item in both cart and form is with new invalid quantity. Only cart object is loaded from database.

**Comment by BabovicT** [comment]
@dpfaffenbauer seems to be working :)

**Review by dpfaffenbauer — APPROVED** [review]

**Pull request merged**